term-utils/write.c

```diff
@@ -129,8 +134,41 @@ static int check_tty(const char *tty, int *tty_writeable, time_t *tty_atime, int
  */
 static int check_utmp(const struct write_control *ctl)
 {
-	struct utmpx *u;
 	int res = 1;
+#if defined(HAVE_LIBSYSTEMD) && HAVE_DECL_SD_SESSION_GET_USERNAME == 1
+	char **sessions_list;
+        int sessions = sd_get_sessions(&sessions_list);
+        if (sessions < 0)
+                errx(EXIT_FAILURE, _("error getting sessions: %s"),
```

Comment:
Maybe `err()` could be taught to handle negative errnos.

Reply:
err() is a libc function and as such will:
1. never see such a change, at least not for glibc, as this would deviate from the standard
2. you need to enhance all libc variants out there and wait until they all got released and are in use
3. err uses errno internal, sd_get_sessions() does not set errno.

I see no way to change err in the required way, the answer would be, for this we have errx. Was long enough myself glibc developer.

Reply:
> err() is a libc function and as such will:

Indeed, it is. I never realized.
I always thought it always comes from `include/c.h`, which defines a local fallback definiton.